activerecord/lib/active_record/test_fixtures.rb

```diff
@@ -189,7 +207,14 @@ def setup_transactional_fixtures
             if pool
               setup_shared_connection_pool
 
-              unless @fixture_connection_pools.include?(pool)
+              # Get the db config for the connection (if it exists)
+              pool_manager = ActiveRecord::Base.connection_handler.send(:connection_name_to_pool_manager)[connection_name]
+              config = pool_manager&.get_pool_config(ActiveRecord::Base.current_role, shard)
```

Comment:
We already have the pool, and there's no way to have a `pool` without a `db_config` so reaching into the private pool manager isn't necessary.

So this can be:

```ruby
config = pool.db_config
```

Reply:
I originally had `config = pool.db_config` but CI failed with some tests reporting that the `pool` object did not have a `db_config` method.

See the results of this previous test run: https://buildkite.com/rails/rails/builds/112075#019248b3-3e1a-411b-8024-e02f6f342f02

```log
Error:
  | TransactionalFixturesOnConnectionNotification#test_transaction_created_on_connection_notification_for_shard:
  | NoMethodError: undefined method `db_config' for #<#<Class:0x00007fde9d2d9e38>:0x00007fde9d274c90>
  | lib/active_record/test_fixtures.rb:212:in `block in setup_transactional_fixtures'
  | /rails/activesupport/lib/active_support/notifications/fanout.rb:133:in `block in finish'
  | /rails/activesupport/lib/active_support/notifications/fanout.rb:24:in `block in iterate_guarding_exceptions'
  | /rails/activesupport/lib/active_support/notifications/fanout.rb:23:in `each'
  | /rails/activesupport/lib/active_support/notifications/fanout.rb:23:in `iterate_guarding_exceptions'
  | /rails/activesupport/lib/active_support/notifications/fanout.rb:121:in `each'
  | /rails/activesupport/lib/active_support/notifications/fanout.rb:132:in `finish'
  | /rails/activesupport/lib/active_support/notifications/fanout.rb:258:in `block in finish_with_values'
  | /rails/activesupport/lib/active_support/notifications/fanout.rb:24:in `block in iterate_guarding_exceptions'
  | /rails/activesupport/lib/active_support/notifications/fanout.rb:23:in `each'
  | /rails/activesupport/lib/active_support/notifications/fanout.rb:23:in `iterate_guarding_exceptions'
  | /rails/activesupport/lib/active_support/notifications/fanout.rb:257:in `finish_with_values'
  | /rails/activesupport/lib/active_support/notifications/fanout.rb:250:in `finish'
  | /rails/activesupport/lib/active_support/notifications/instrumenter.rb:64:in `instrument'
  | test/cases/fixtures_test.rb:1156:in `block in fire_connection_notification'
```

Looks like not all instances coming from `ActiveRecord::Base.connection_handler.retrieve_connection_pool(connection_name, shard: shard)` have that method/field?

Reply:
That test is using a fake pool object, so it'll presumably need the extra method added if we're not expecting it to be there.

Reply:
Resolved by extending the fake pool objects with the correct method. Thanks for the pointers!